File: C++/old/3 Longest Substring Without Repeating Characters/code1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left = 0;
        unordered_set<char> chars;
        int length = 0;
        for(int right = 0; right < s.length(); right++)
        {
            while(chars.find(s[right]) != chars.end())
            {
                chars.erase(s[left]);
                left++;
            }
            chars.insert(s[right]);
            if(length < chars.size()) length = chars.size();
        }
        return length;
    }
};

File: Python/125 Valid Palindrome/code1.py
def solution(s):
    left = 0
    right = len(s) - 1
    s = s.lower()
    while left < right:
        while left < len(s) and not s[left].isalnum(): left += 1
        while right >= 0 and not s[right].isalnum(): right -= 1
        if left >= right: break
        if s[left] != s[right]: return False
        left += 1
        right -= 1
    return True
